### ISSUE TITLE
fast-xml-parser not listed as a dependency of @aws-sdk/client-s3

### ISSUE THREAD

**Issue by lineape** [aws/aws-sdk-js-v3#5866]
### Checkboxes for prior research

- [X] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest)
- [X] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js).
- [X] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-js-v3/issues) and didn't find any solution.

### Describe the bug

@aws-sdk/client-s3 3.529.0 no longer lists fast-xml-parser as a dependency, even though it's used at runtime (and is not bundled in the compiled javascript).

This causes module not found errors.

### SDK version number

@aws-sdk/client-s3@3.529.0

### Which JavaScript Runtime is this issue in?

Node.js

### Details of the browser/Node.js/ReactNative version

v18.19.1

### Reproduction Steps

```
mkdir test && cd test
npm init -y && npm add @aws-sdk/client-s3
echo 'new require("@aws-sdk/client-s3").S3Client();' > test.js
node test.js
```

### Observed Behavior

node:internal/modules/cjs/loader:1137
  throw err;
  ^

Error: Cannot find module 'fast-xml-parser'

### Expected Behavior

No errors

### Possible Solution

Re-add fast-xml-parser to dependencies or bundle fast-xml-parser with build.

**Comment by kuhe** [comment]
Hi, I'm sorry for releasing this bug. Our validation scripts have a gap in checking external dependencies, which I'll work on a patch for. 

As a workaround, please use the previous version or add `fast-xml-parser@4.2.5` to your application temporarily.

**Comment by diegodleon** [comment]
No prob! Thanks for the quick fix, any ETAs on the release of the fix? @kuhe

**Comment by trivikr** [comment]
We're aiming to get this issue fixed in future release https://github.com/aws/aws-sdk-js-v3/releases/tag/v3.529.1
Expected time of release as of this comment is around 5 PM Pacific, i.e. in two hours.

**Comment by tdq132** [comment]
I ran into the same issue when using `client-ec2` via a CDK Custom Resource. 
Workaround for me was to add this to `cdk.json` context: `"@aws-cdk/customresources:installLatestAwsSdkDefault": false`

**Comment by kuhe** [comment]
The fix was released in https://www.npmjs.com/package/@aws-sdk/client-s3/v/3.529.1 and other client packages of the same version.

**Comment by trivikr** [comment]
> We're aiming to get this issue fixed in future release [`v3.529.1` (release)](https://github.com/aws/aws-sdk-js-v3/releases/tag/v3.529.1)

The `v3.529.1` is out.

I verified using minimal repro that the "module not found" error is no longer thrown.
```console
mkdir test && cd test
npm init -y && npm add @aws-sdk/client-s3
echo 'import {S3Client} from "@aws-sdk/client-s3"; const client = new S3Client();' > test.mjs
node test.mjs
```

If you're using the latest versions of AWS SDK for JavaScript v3, there's no action needed. The issue will get fixed when you run install next time. If you've pinned SDK version to v3.529.0 in your package.json or lockfile, please update it.
Any workaround added, like explicitly adding `fast-xml-parser` dependencies, can be removed after upgrading to v3.529.1

Thank you for your patience. We apologize for the inconvenience caused.
We're adding validation script in https://github.com/aws/aws-sdk-js-v3/pull/5871 to ensure SDK is not released with a missing dependency.

**Comment by ddolcimascolo** [comment]
Thx for fixing this quickly, and for your work on this great library!

Cheers,
David
